packages/ckeditor5-utils/src/keyboard.js

```diff
@@ -108,22 +114,20 @@ export function parseKeystroke( keystroke ) {
  * @returns {String} Keystroke text specific for the environment.
  */
 export function getEnvKeystrokeText( keystroke ) {
-	if ( !env.isMac ) {
-		return keystroke;
-	}
+	let keystrokeCode = parseKeystroke( keystroke );
+
+	const modifiersToGlyphs = Object.entries( env.isMac ? modifiersToGlyphsMac : modifiersToGlyphsNonMac );
+
+	const modifiers = modifiersToGlyphs.reduce( ( modifiers, [ name, glyph ] ) => {
+		if ( ( keystrokeCode & keyCodes[ name ] ) != 0 ) {
+			keystrokeCode &= ~keyCodes[ name ];
```

Comment:
I am wondering if we should add a comment with a short explanation/description of what actually is happening here. It's not totally obvious at first and this construction is not used often in our codebase.

Reply:
Also, bloody hell, this is sooo clean! 🎉

Reply:
Yeah, that's some nice programming porn 👏 But I second the request for comments because unless you run a debugger, it's magic.